### ISSUE TITLE
Add option for consistent-return to support methods which throw

### ISSUE THREAD

**Issue by emmettnicholas** [eslint/eslint#5269]
My code base makes use of this pattern:

```javascript
function getFooName(foo) {
    switch (foo) {
        case Foo.ABC:
            return 'abc';

        case Foo.DEF:
            return 'def';

        default:
            throwError('invalid foo: ' + foo);
    }
}

function throwError(name) {
    throw new Error(name);
}
```

This understandably violates the "consistent-return" rules since ESLint doesn't know that `throwError` throws.

Locally I forked consistent-return to accept a "throwPattern" option, e.g.:

     "consistent-return": [2, {"throwPattern": "^throwError"}],

I'm curious to know whether this is worth polishing and submitting a PR for.

**Comment by platinumazure** [comment]
Just out of curiosity, what if you returned an error from that function and explicitly threw?

Something like:

```js
function getFooName(foo) {
    switch (foo) {
        case Foo.ABC:
            return 'abc';

        case Foo.DEF:
            return 'def';

        default:
            throw getError('invalid foo: ' + foo);
    }
}

function getError(name) {
    return new Error(name);
}
```

**Comment by emmettnicholas** [comment]
That would work. In fact refactoring the code, instead of augmenting ESLint, may be the superior option now that I think about it more.

**Comment by emmettnicholas** [comment]
After thinking about it some more, the code in my original post feels like an anti-pattern; I'm going to close this.
